test/gen-server/ApiServer.ts

```diff
@@ -2164,397 +2164,414 @@ describe('ApiServer', function() {
+        assert.equal(resp.status, 200);
+        assert.isArray(resp.data);
+        assert.lengthOf(resp.data, 2);
+      });
 
-    it('Endpoint GET /api/service-accounts is operational', async function() {
-      const body = {
-        label: "A small service for the chimpy",
-        description: "A big service for robotkind",
-        endOfLife:"2042-07-21",
-      };
-      const body2 = {
-        label: "More service",
-        description: "More robots",
-        endOfLife:"2042-07-22",
-      };
-      await axios.post(`${homeUrl}/api/service-accounts/`, body, chimpy);
-      await axios.post(`${homeUrl}/api/service-accounts/`, body2, chimpy);
-      const resp = await axios.get(`${homeUrl}/api/service-accounts/`, chimpy);
-      assert.equal(resp.status, 200);
-      assert.isArray(resp.data);
-      assert.lengthOf(resp.data, 2);
+      it("returns 401 when user is anonymous", async function() {
+        const resp = await axios.get(`${homeUrl}/api/service-accounts`, nobody);
+        assert.equal(resp.status, 401);
+      });
     });
 
-    it("Endpoint GET /api/service-accounts returns 404 when user don't own any service account", async function() {
-      const resp = await axios.get(`${homeUrl}/api/service-accounts`, chimpy);
-      assert.equal(resp.status, 404);
-    });
+    describe('Endpoint GET /api/service-accounts/{saId}', function () {
+      it('is operational', async function() {
+        const {login: serviceLogin} = await createServiceAccount(SERVICE_ACCOUNT_BODY);
+        const expectedBody = {
+          ...SERVICE_ACCOUNT_BODY,
+          endOfLife: `${SERVICE_ACCOUNT_BODY.endOfLife}T00:00:00.000Z`,
+          hasValidKey: true
+        };
+        const resp = await axios.get(`${homeUrl}/api/service-accounts/${serviceLogin}`, chimpy);
+        assert.equal(resp.status, 200);
+        assert.isObject(resp.data);
+        assert.deepEqual(resp.data, {login:serviceLogin, ...expectedBody});
+      });
 
-    it('Endpoint GET /api/service-accounts/{saId} is operational', async function() {
-      const body = {
-        label: "A small service for the chimpy",
-        description: "A big service for robotkind",
-        endOfLife:"2042-07-21",
-      };
-      const resp = await axios.post(`${homeUrl}/api/service-accounts/`, body, chimpy);
-      const serviceLogin = resp.data.login;
-      const expectedBody = {
-        ...body,
-        endOfLife: `${body.endOfLife}T00:00:00.000Z`,
-        hasValidKey: true
-      };
-      const resp2 = await axios.get(`${homeUrl}/api/service-accounts/${serviceLogin}`, chimpy);
-      assert.equal(resp2.status, 200);
-      assert.isObject(resp2.data);
-      assert.hasAllKeys(resp2.data, ["login", "label", "description", "endOfLife", "hasValidKey"]);
-      assert.deepEqual(resp2.data, {login:serviceLogin, ...expectedBody});
-    });
+      it('returns 403 for non-owned service accounts {saId}', async function() {
```

Comment:
Refactoring would be ideal, most of these tests have the same logic in the different describes